Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs

```diff
@@ -564,12 +587,35 @@ bookingCancelledReqHandler ValidatedBookingCancelledReq {..} = do
   unless (booking.status == DRB.CANCELLED) $ void $ QRB.updateStatus booking.id DRB.CANCELLED
   whenJust mbRide $ \ride -> void $ do
     unless (ride.status == DRide.CANCELLED) $ void $ QRide.updateStatus ride.id DRide.CANCELLED
+  fork "Cancellation Settlement" $ do
+    whenJust cancellationFee $ \fee -> do
+      riderConfig <- QRiderConfig.findByMerchantOperatingCityId booking.merchantOperatingCityId >>= fromMaybeM (InternalError "RiderConfig not found")
+      merchant <- CQM.findById booking.merchantId >>= fromMaybeM (MerchantNotFound booking.merchantId.getId)
+      person <- QP.findById booking.riderId >>= fromMaybeM (PersonNotFound booking.riderId.getId)
+      merchantCity <- CQMOC.findById booking.merchantOperatingCityId >>= fromMaybeM (MerchantOperatingCityNotFound booking.merchantOperatingCityId.getId)
+      mobileNumber <- mapM decrypt person.mobileNumber >>= fromMaybeM (PersonFieldNotPresent "mobileNumber")
```

Comment:
MobileNumber is not the necessary field as it will not be there in case of Bridge. Why do we need it?

Reply:
We have made mobile number as mandatory in bridge.
Should I not throw error if not found? Just logError